app/helpers/compute_resources_vms_helper.rb

```diff
@@ -263,23 +263,24 @@ def vm_import_action(vm, html_options = {})
         :type => 'unmanaged'),
       html_options
     )
-
     import_managed_link + import_unmanaged_link
   end
 
-  def vm_associate_action(vm)
-    display_link_if_authorized(
-      _("Associate VM"),
-      hash_for_associate_compute_resource_vm_path(
-        :compute_resource_id => @compute_resource,
-        :id => vm.identity
-      ).merge(
-        :auth_object => @compute_resource,
-        :permission => 'edit_compute_resources'),
-      :title => _("Associate VM to a Foreman host"),
-      :method => :put,
-      :class => "btn btn-default"
-    )
+  def vm_associate_action(vm, isdropdown: false)
+    if @compute_resource.supports_host_association?
+      display_link_if_authorized(
```

Comment:
Can you extract this to a helper method called `associate_vm_button(vm, klass)` and use this method in `vm_associate_action` and `ovirt_vms_data`?

I believe this helps a lot to declutter the code.
Sorry for being picky here, but this is not the prettiest code to begin with. I think we should make sure we clean it up a little when we have the chance.

Reply:
@timogoebel Can you clarify the extraction you are suggesting? I'm not sure I followed

Reply:
Something like this:

```ruby
def vm_associate_action(vm)
  associate_vm_link(vm, link_class: "btn btn-default")
end


def associate_vm_link(vm, link_class = '')
  display_link_if_authorized(
      _('Associate VM'),
      hash_for_associate_compute_resource_vm_path(
        :compute_resource_id => @compute_resource,
        :id => vm.identity
      ).merge(
        :auth_object => @compute_resource,
        :permission => 'edit_compute_resources'),
      :title => _("Associate VM to a Foreman host"),
      :method => :put,
      :class => link_class
    )
end

# and in ovirt_vms_data
vm_import_action(vm), vm_associate_link(vm),
```